test/hotspot/jtreg/serviceability/dcmd/vm/VMInspectTest.java

```diff
@@ -0,0 +1,219 @@
+
+    //  tid=0x0000153418029c20
+    static Pattern thread_id_line =
+        Pattern.compile(" tid=0x(\\p{XDigit}+) ");
+
+    public void run(CommandExecutor executor) throws ClassNotFoundException {
+        DcmdTestClass test = new DcmdTestClass();
+        test.work();
+        BigInteger ptr = null;
+        OutputAnalyzer output = null;
+
+        // Testing VM.inspect requires UnlockDiagnosticVMOptions or a debug JVM.
+        // This test runs with a System Property set as a hint whether UnlockDiagnosticVMOptions is set.
+        boolean enabled = Platform.isDebugBuild() || Boolean.getBoolean("vminspect.enabled");
+        System.out.println("VM.inspect should be enabled = " + enabled);
+        if (!enabled) {
+            // Use any pointer, command should be refused:
+            output = executor.execute("VM.inspect 0x0");
+            output.shouldContain("-XX:+UnlockDiagnosticVMOptions is required");
+            return; // no more testing
+        }
+
+        // Tests where enabled:
+        output = executor.execute("help");
+        output.shouldNotContain("VM.inspect"); // VM.inspect is not promoted in help
+        output = executor.execute("help VM.inspect");
+        output.shouldContain("Syntax : VM.inspect"); // but help is available
+
+        // Test VM.inspect:
+        testFind(executor);
+    }
+
+    public void testFind(CommandExecutor executor) {
+        boolean testMisaligned = true;
+        OutputAnalyzer output = executor.execute("VM.inspect");
+        output.shouldContain("Usage: VM.inspect");
+        // Find and test a thread id:
+        OutputAnalyzer threadPrintOutput = executor.execute("Thread.print");
+        BigInteger ptr = findPointer(threadPrintOutput, thread_id_line, 1);
+        output = executor.execute("VM.inspect " + pointerText(ptr));
+        output.shouldContain(" is a thread");
+        // verbose shows output like:
+        // "main" #1 [17235] prio=5 os_prio=0 cpu=1265.79ms elapsed=6.12s tid=0x000014e37802bd80 nid=17235 in Object.wait()  [0x000014e3817d4000]
+        //    java.lang.Thread.State: WAITING (on object monitor)
+        // Thread: 0x000014e37802bd80  [0x4353] State: _running _at_poll_safepoint 0
+        // ...
+        // Also a debug vm shows: JavaThread state: _thread_blocked
+        // ...
+        output = executor.execute("VM.inspect -verbose " + pointerText(ptr));
+        output.shouldContain("java.lang.Thread.State: WAITING");
+
+        // Known bad pointers:
+        output = executor.execute("VM.inspect 0x0");
+        output.shouldContain("address not safe");
+
+        output = executor.execute("VM.inspect -1");
+        output.shouldContain("address not safe");
+
+        // Find and test a Java Object:
+        threadPrintOutput = executor.execute("Thread.print");
+        ptr = findPointer(threadPrintOutput, waiting_on_mylock, 1);
+        output = executor.execute("VM.inspect " + pointerText(ptr));
+        System.out.println(output);
```

Comment:
Nit: May I ask you to add empty lines after the lines 101 and 123 to make the code more readable?